src/llm/http_llm_calculator.cc

```diff
@@ -131,17 +168,71 @@ class OpenAIChatCompletionsRequest {
                 return false;
         }
 
-        // presence_penalty: float; optional - defaults to 0
-        it = this->doc.FindMember("presence_penalty");
+        // TODO: Supported by OpenAI and vLLM, however unsupported by CB lib
+        // // frequency_penalty: float; optional - defaults to 0
+        // it = this->doc.FindMember("frequency_penalty");
+        // if (it != this->doc.MemberEnd()) {
+        //     return false;  // TODO: Unsupported by CB
+        //     if (!it->value.IsDouble())
+        //         return false;
+        //     this->frequencyPenalty = it->value.GetDouble();
+        //     if (this->frequencyPenalty < -2.0f || this->frequencyPenalty > 2.0f)
+        //         return false;
+        // }
+
+        // TODO: Supported by OpenAI and vLLM, however unsupported by CB lib
+        // // presence_penalty: float; optional - defaults to 0
+        // it = this->doc.FindMember("presence_penalty");
+        // if (it != this->doc.MemberEnd()) {
+        //     return false;  // TODO: Unsupported by CB
+        //     if (!it->value.IsDouble())
+        //         return false;
+        //     this->presencePenalty = it->value.GetDouble();
+        //     if (this->presencePenalty < -2.0f || this->presencePenalty > 2.0f)
+        //         return false;
+        // }
+
+        // repetition_penalty: float; optional - defaults to 1.0
+        // Extension, unsupported by OpenAI API, however supported by vLLM and CB lib
+        it = this->doc.FindMember("repetition_penalty");
         if (it != this->doc.MemberEnd()) {
             if (!it->value.IsDouble())
                 return false;
-            this->presencePenalty = it->value.GetDouble();
-            if (this->presencePenalty < -2.0f || this->presencePenalty > 2.0f)
+            this->repetitionPenalty = it->value.GetDouble();
+            // TODO: Validate?
```

Comment:
I don't think so. For now we should probably simply catch errors from cb lib calls and eventually such validation should be done on the library side I think (non-OVMS users can also specify bad generation parameters when using it as a standalone)

Reply:
Validation/error handling is on separate branch of @michalkulakowski